Feat/#7 place - 카테고리, 인플루언서 필터링을 적용시켰어요!

dong-yxxn opened the pull request:
### ✨ 작업 내용

- 카테고리, 인플루언서 필터링을 적용시켰습니다.
- 관련된 테스트코드도 작성하였습니다.
---

### ✨ 참고 사항

---

### ⏰ 현재 버그
- controller 테스트 코드가 미완성이라 작동시키면 에러납니다.
---

### ✏ Git Close

dong-yxxn committed:
[feat]: 장소 조회에 카테로리, 인플루언서 필터 적용

- 장소조회에 Category, Influencer 필터를 적용시켰습니다.
- PlacesCommand에 PlacesFilterParamsCommand를 추가하여 필터링 정보를 전송하였습니다.
- Repository에 대한 테스트코드 작성하였습니다.

- 관련 이슈: #7

dong-yxxn committed:
[refactor] PlaceInfo에 인플루언서 정보 적용

- PlaceInfo의 influencerName을 Null처리에서 name을 조회해서 반영하도록 했습니다.

관련 이슈: #7

dong-yxxn committed:
[fix] 충돌 해결

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -1,33 +1,76 @@
+
+        // 필터 값이 있을 경우에만 split 처리
+        if (placesFilterParamsCommand.categories() != null
+            && !placesFilterParamsCommand.categories().isEmpty()) {
```
empty를 검증해주는 메서드를 DTO에 만드는 게 가독성이 더 좋을 것 같아요!

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -1,33 +1,76 @@
+        }
+
+        if (placesFilterParamsCommand.influencers() != null
+            && !placesFilterParamsCommand.influencers().isEmpty()) {
```
여기도 마찬가지로 따로 메서드를 만드는게 좋을 것 같아요

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -1,33 +1,76 @@
+            String influencerName = videoRepository.findByPlaceId(place.getId()).getInfluencer()
+                .getName();
+            // PlaceInfo 객체 생성
+            return PlaceInfo.of(place, influencerName);
```
videoRepository에 개별 조회 방식 말고도 다른 메서드 작명을 통해서 여러 번 조회하지 않고 한 번에 가져올 수 있지 않을까요?

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -1,33 +1,76 @@
+            placesCoordinateCommand.latitude(),
+            placesCoordinateCommand.longitude(),
+            categoryFilters != null && !categoryFilters.isEmpty() ? categoryFilters : null,
+            influencerFilters != null && !influencerFilters.isEmpty() ? influencerFilters : null,
```
윗 부분과 어떤  점이 다른거죠? 다르지 않다면 이 방식으로 하는게 낫지 않을까요?

BaeJunH0 commented on src/main/java/team7/inplace/video/persistence/VideoRepository.java:
```diff
@@ -1,15 +1,18 @@
+
     List<Video> findVideosByInfluencerIdIn(List<Long> influencerIds);
 
+    Video findByPlaceId(Long placeId);
```
위의 메서드 처럼 매개변수로 List를 받는게 어떨까요

BaeJunH0 commented on src/test/java/team7/inplace/place/repository/PlaceRepositoryTest.java:
```diff
@@ -2,32 +2,43 @@
+import team7.inplace.video.domain.Video;
 
 @DataJpaTest
+@Transactional
```
필요한가요?

BaeJunH0 commented on src/test/java/team7/inplace/place/repository/PlaceRepositoryTest.java:
```diff
@@ -48,17 +59,154 @@ class PlaceRepositoryTest {
+        entityManager.persist(video1);
+    }
+
+    @Test
+    public void 거리기반_장소조회_테스트() {
```
메서드 명을 한글로 하기 보다는, 메서드 명을 test1, 2, 3으로 하더라도 @Display 사용해서 이름을 나타내는 게 나을 것 같아요

BaeJunH0 commented on src/test/java/team7/inplace/video/service/VideoServiceTest.java:
```diff
@@ -15,14 +15,20 @@
 
 @ExtendWith(MockitoExtension.class)
 public class VideoServiceTest {
+
```
공백이 있어야 하나요?

BaeJunH0 replied:
어라 제 부분이네요....

BaeJunH0 reviewed (changes requested):
전체적으로 잘 해주셨습니다, 특히 테스트 코드 열심히 작성해주신 것 같아요, 근데 토큰 관련 처리에 대해 말씀하셨던 거 같은데, 해당 부분이 컨트롤러에 빠졌네요 아직 못하신 건가요?

dong-yxxn committed:
[refactor] 리뷰 반영

- PlaceService: empty검증 메서드 DTO에 작성함
-- videoRepo에서 placeId 한번에 조회하도록 메서드 수정
-- PlaceRepoTest: Transactional 제거
-- 테스트 이름 영어로 수정

관련 이슈: #7

dong-yxxn committed:
[refactor]: 테스트 패키지 이름 수정

- application -> presentation으로 이름 수정
- repository -> persistence로 이름 수정

관련 이슈: #7

dong-yxxn committed:
[refactor] findByPlaceIds -> findByPlaceIdIn으로 변경

- videoRepository에서 제목과같이 변경 후, PlaceService에 적용하였습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor & fix] Place 엔티티 수정, test 코드 오류 수정

- offdayList 추가를 위해 PlaceCloseTime VO를 추가하고, PlaceTime VO를 PlaceOpenTime으로 변경해서 구분지었습니다.
- PlaceServiceTest: 기존 place의 Id가 null로 지정되어있어 influencer 이름이 제대로 저장되지 않는 문제를 발견함
-- 조치완료

관련 이슈: #7

suhyeon7497 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -1,33 +1,89 @@
+
+        // PlaceInfo 리스트를 Page로 변환하여 반환
+        return new PageImpl<>(placeInfos, placesPage.getPageable(), placesPage.getTotalElements());
     }
```
method 길이가 너무 길어진것같은데, refactoring이 필요하지않을까 싶네요.

suhyeon7497 commented on src/main/java/team7/inplace/place/presentation/PlaceController.java:
```diff
@@ -35,10 +36,13 @@ public ResponseEntity<PlacesResponse> getPlaces(
         @RequestParam(defaultValue = "0") int page,
         @RequestParam(defaultValue = "10") int size
     ) {
+        // 위치기반 조회
```
@PageableDefault도 존재하는데 사용해보시는 거 어떨까요?

suhyeon7497 reviewed (approved):
refactoring 부분이랑, test코드를 고쳐주시면 될거같아요!
추가적으로 생성자를 꼭 public으로만 해야하는 지 생각해보시면 좋을거같아요.
